akka-docs/rst/scala/http/introduction.rst

```diff
@@ -10,23 +10,105 @@ You get to pick the API level of abstraction that is most suitable for your appl
 This means that, if you have trouble achieving something using a high-level API, there's a good chance that you can get
 it done with a low-level API, which offers more flexibility but might require you to write more application code.
 
+DSL for HTTP servers
+--------------------
+The high-level, routing API of Akka HTTP provides a DSL to describe HTTP "routes" and how they should be handled.
+Each route is composed of one or more level of ``Directive`` s that narrows down to handling one specific type of
+request.
+
+For example one route might start with matching the ``path`` of the request, only matching if it is "/hello", then
+narrowing it down to only handle HTTP ``get`` requests and then ``complete`` those with a Scala XML literal, which
+will be sent back as a HTTP OK with the XML as HTML in the response body.
+
+Transforming request and response bodies between over-the-wire formats and objects to be used in your application is
+done separately from the route declarations, and is pulled in implicitly using the "magnet" pattern. This means that
+you can basically ``complete`` a request with any kind of value, a as long as there is a marshaller available in scope.
+
+For many simple Scala types and some of the most common ways to complete a response marshallers are available out of
+the box with no additional import needed, for more complex types, such as JSON or XML a dependency containing such
+marshallers must be added (See :ref:`akka-http-spray-json` and :ref:`akka-http-xml-marshalling`)
+
+The ``Route`` created using the Route DSL is then "bound" to a port to start serving HTTP requests:
+
+.. includecode2:: ../code/docs/http/scaladsl/HttpServerExampleSpec.scala
+   :snippet: minimal-routing-example
+
+A common use case is to reply to a request using a model object having the marshaller transform it into JSON. In
+this case shown by two separate routes where one returns an ``Item`` that is marshalled into JSON and one accepts
+JSON that is unmarshalled into an ``Order``.
+
+The logic for the marshalling and unmarshalling is provided by the "spray-json" library (Details how to use that here
+:ref:`akka-http-spray-json`) as two methods creating reflection based implementations:
```

Comment:
That's once for each Format, not "during marshalling" so I always think of spray-json as "no reflection after initial setup" => pretty much no reflection as for the parsing. Or am I missing something since some years :)